GTG/gtk/general_preferences.py

```diff
@@ -0,0 +1,290 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Getting Things GNOME! - a personal organizer for the GNOME desktop
+# Copyright (c) 2008-2013 - Lionel Dricot & Bertrand Rousseau
+#
+# This program is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program.  If not, see <http://www.gnu.org/licenses/>.
+# -----------------------------------------------------------------------------
+
+""" The Preferences Window for configuring GTG """
+
+import os
+import shutil
+import locale
+
+from gi.repository import Gtk, Gdk
+from xdg.BaseDirectory import xdg_config_home
+
+from GTG.core.dirs import UI_DIR
+from GTG.core.translations import _
+from GTG.tools import shortcut
+
+AUTOSTART_DIRECTORY = os.path.join(xdg_config_home, "autostart")
+AUTOSTART_FILE = "gtg.desktop"
+AUTOSTART_PATH = os.path.join(AUTOSTART_DIRECTORY, AUTOSTART_FILE)
+locale.setlocale(locale.LC_ALL, 'sk_SK.utf8')
+
+
+def enable_gtg_autostart():
+    """ Enable autostart
+
+    Firstly, locate gtg.desktop file. Then link it in AUTOSTART_FILE.
+    On Windows, there is no os.symlink, just copy the file. """
+    desktop_file_path = None
+    this_directory = os.path.dirname(os.path.abspath(__file__))
+    for path in ["../..", "../../../applications",
+                 "../../../../../share/applications"]:
+        fullpath = os.path.join(this_directory, path, AUTOSTART_FILE)
+        fullpath = os.path.normpath(fullpath)
+        if os.path.isfile(fullpath):
+            desktop_file_path = fullpath
+            break
+
+    if desktop_file_path:
+        if not os.path.exists(AUTOSTART_DIRECTORY):
+            os.mkdir(AUTOSTART_DIRECTORY)
+
+        # If the path is a symlink and is broken, remove it
+        if os.path.islink(AUTOSTART_PATH) and \
+                not os.path.exists(os.path.realpath(AUTOSTART_PATH)):
+            os.unlink(AUTOSTART_PATH)
+
+        if os.path.isdir(AUTOSTART_DIRECTORY) and \
+                not os.path.exists(AUTOSTART_PATH):
+            if hasattr(os, "symlink"):
+                os.symlink(desktop_file_path, AUTOSTART_PATH)
+            else:
+                shutil.copyfile(desktop_file_path, AUTOSTART_PATH)
+
+
+def disable_gtg_autostart():
+    """ Disable autostart, removing the file in autostart_path """
+    if os.path.isfile(AUTOSTART_PATH):
+        os.remove(AUTOSTART_PATH)
```

Comment:
Can you please move this code into GTG.tools.autostart? This isn't core GTG and it is nicely separate from the rest of the code.